tests/test_data.py

```diff
@@ -0,0 +1,221 @@
+    assert positive_pairs[5].first.identity == 'TEST-2'
+    assert positive_pairs[6].first.identity == 'TEST-3'
+    assert positive_pairs[7].first.identity == 'TEST-4'
+
+
+def test_make_pairs_positive_only_fixed_n(dummy_images):
+    n = 8
+    pairs = make_pairs(dummy_images, same=True, n=n)
+    positive_pairs = [p for p in pairs if p.same_identity]
+    assert len(positive_pairs) == len(pairs) == 8
+
+
+def test_make_pairs_positive_only_fixed_n_odd(dummy_images):
+    n = 7
+    pairs = make_pairs(dummy_images, same=True, n=n)
+    positive_pairs = [p for p in pairs if p.same_identity]
+    assert len(positive_pairs) == len(pairs) == 7
+
+
+def test_make_pairs_positive_only_large_n(dummy_images):
+    n = 10000
+    pairs = make_pairs(dummy_images, same=True, n=n)
+    positive_pairs = [p for p in pairs if p.same_identity]
+    assert len(positive_pairs) == len(pairs) == 8
+
+
+def test_make_pairs_negative_only_no_n(dummy_images):
+    pairs = make_pairs(dummy_images, same=False, n=None)
+    negative_pairs = [p for p in pairs if not p.same_identity]
+    assert len(negative_pairs) == len(pairs) == 94
+    assert negative_pairs[0].first.identity == 'TEST-1'
+    assert negative_pairs[24].first.identity == 'TEST-2'
+    assert negative_pairs[48].first.identity == 'TEST-3'
+    assert negative_pairs[66].first.identity == 'TEST-4'
+    assert negative_pairs[84].first.identity == 'TEST-5'
+
+
+def test_make_pairs_negative_only_fixed_n(dummy_images):
+    n = 48
+    pairs = make_pairs(dummy_images, same=False, n=n)
+    negative_pairs = [p for p in pairs if not p.same_identity]
+    assert len(negative_pairs) == len(pairs) == n
+
+
+def test_make_pairs_negative_only_fixed_n_odd(dummy_images):
+    n = 47
+    pairs = make_pairs(dummy_images, same=False, n=n)
+    negative_pairs = [p for p in pairs if not p.same_identity]
+    assert len(negative_pairs) == len(pairs) == n
+
+
+def test_make_pairs_negative_only_large_n(dummy_images):
+    n = 10000
+    pairs = make_pairs(dummy_images, same=False, n=n)
+    negative_pairs = [p for p in pairs if not p.same_identity]
+    assert len(negative_pairs) == len(pairs) == n
+
+
+################
+# `LfwDataset` #
+################
+
+# TODO: add tests for LfwDataset.
+
+
+##################
+# `EnfsiDataset` #
+##################
+
+@skip_if_missing(EnfsiDataset)
+def test_enfsi_dataset_has_correct_num_images(enfsi_all):
+    assert len(enfsi_all.images) == 270
+
+
+@skip_if_missing(EnfsiDataset)
+def test_enfsi_dataset_has_correct_num_pairs(enfsi_all):
+    assert len(enfsi_all.pairs) == 135
+    assert all([a.meta['idx'] == b.meta['idx'] for a, b in enfsi_all.pairs])
+
+
+###############
+# `ForenFace` #
+###############
+
+@skip_if_missing(ForenFaceDataset)
+def test_forenface_dataset_has_correct_num_images(forenface):
+    assert len(forenface.images) == 2476
+
+
+@skip_if_missing(ForenFaceDataset)
+def test_forenface_dataset_has_correct_num_pairs(forenface):
+    assert len(forenface.pairs) == 60798
```

Comment:
wil je deze hard hierin zetten? Er zijn geen natuurlijke paren toch?

Reply:
Goed punt ja, ik haal deze eruit

- [ ] Remove unit test for pairs in forenface